src/hooks/useScrollPercent.ts

```diff
@@ -1,4 +1,4 @@
-import { useCallback, useMemo, useState } from 'react'
+import { useCallback, useMemo, useState } from 'preact/hooks'
 import useThrottle from 'hooks/useThrottle'
 
 export default function () {
```

Comment:
We don't need two different hooks with almost same logic, we can detect if user is not at the top with this one as well